### ISSUE TITLE
[Container Tabs] Loss of tab parent when using containers with contextual menu

### ISSUE THREAD

**Issue by TheSweetLily** [piroor/treestyletab#1549]
## Short description
Say you want to search for something on Google; you highlight the text and press "search Google for..". 
New tab will be opened with the previous tab you searched from as a parent. 
Everything works as expected. 

Now, adding Google to an isolated container causes the tab to lose its parent, so it just becomes a new tab. 

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Install [Firefox Multi-Account Containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers)
 4. Open Google.com (Or whichever top level domain opens when you search with Google)
 5. Assign Google.com to always open in a new container (e.g. work container)
 6. Open Google.com and confirm you always want Google.com to always open in the container
 7. Open a new tab/previous tab to highlight something and search for it on Google with the context menu "Search Google for "..." ". 

## Expected result
The tab with the search has the tab the search originated from as its parent.
![](https://i.imgur.com/PSoNd5m.png)

## Actual result
The tab loses its parent and is on its own.
![](https://i.imgur.com/8fQ97OY.png)

## Environment

 * Platform (OS): W10
 * Version of Firefox: Latest nightly.
 * Version (or revision) of Tree Style Tab: 2.2.9

**Comment by piroor** [comment]
Thanks. By recent commits tabs opened with position by other addons are treated more correctly and tabs replaced by Firefox Multi-Account Containers are attached instead of the original tab.

**Comment by TheSweetLily** [comment]
Hm, this is weird.. 
I installed your latest Nightly from your website, and it worked on the first try, but upon trying it again it stopped working.

To isolate any addons or settings possibly breaking it I created a new profile and retraced my steps from above.. I cannot get it to work again for some reason. 
___
Trying it again while writing this on this tab seems to have it working again, hm.. 
However, trying it on a new tab (reddit.com) and searching on Google from there doesn't have it working.
And now it suddenly works again on the same tab after after a few minutes, even when searching on the same exact elements that didn't work before.

It seems that it takes some time before it saves some sort of info or something?
___
Trying this on a few tabs, and it seems to be the same every time whenever it doesn't work for a tab. (Some tabs it works right away.)
I just let the tab sit for a few minutes, and then it starts working. 
___
Anyway I can diagnose what the cause is?
